dev/test/write-batch.ts

```diff
@@ -76,6 +78,24 @@ describe('set() method', () => {
     nullObject.bar = 'ack';
     writeBatch.set(firestore.doc('sub/doc'), nullObject);
   });
+
+  it('requires toFirestoreFromMerge() for Partial usage', () => {
+    const converter = postConverter;
+    delete converter.toFirestoreFromMerge;
+    const ref = firestore.doc('sub/doc').withConverter(converter);
+    expect(() =>
+      writeBatch.set(ref, {title: 'foo'} as Partial<Post>, {merge: true})
+    ).to.throw(
+      'toFirestoreFromMerge() must be defined to use set() with `merge` or `mergeFields`.'
+    );
+  });
+
+  it('requires SetOptions to use partials', () => {
+    const ref = firestore.doc('sub/doc').withConverter(postConverter);
+    expect(() =>
+      (writeBatch as InvalidApiUsage).set(ref, {title: 'foo'} as Partial<Post>)
+    ).to.throw('Value for argument "data" is not a valid Firestore document.');
```

Comment:
How does you code know that  `{title: 'foo'}` is not a `Post`?

Reply:
Oops forgot to remove.

Reply:
Oh, does this throw because author is now `undefined`? I feel like that it is likely more a test of the converter. I suggest to rework is as a converter test or to remove.

Reply:
Removed the test, since we already have other tests that check for undefined values.